File: tests/ollama.py
import requests
import json
import os

def get_ollama_status():
    try:
        response = requests.get("http://localhost:11434/api/tags", timeout = 2)
        if response.status_code == 200:
            return "Ollama System Online"
        else: 
            return f"Error occured while connecting to Ollama: {response.status_code}"
    except requests.exceptions.RequestException as e:
        return f"Ollama is not responding: {str(e)}"

File: go/internal/health/ollama.go
package health

import (
	"fmt"
	"net/http"
	"os"
)

// CheckOllamaHealth verifies if the Ollama service is running
func CheckOllamaHealth() {
	response, err := http.Get("http://localhost:11434/api/tags")
	if err != nil {
		fmt.Println(err.Error())
		fmt.Println("Ollama healthcheck failed. Exiting...")
		os.Exit(1)
	}
	defer response.Body.Close() // Properly close the response body
	
	if response.StatusCode == 200 {
		fmt.Println("Ollama system online")
	} else {
		fmt.Printf("Ollama returned unexpected status code: %d\n", response.StatusCode)
		os.Exit(1)
	}
}
